### PR TITLE
ah - made the UI to create a list

### LINKED ISSUE #124: Create UserList UI
Create a user list ui

Acceptance Criteria:
- [ ] Create a user list page/component
- [ ] Able to set the Name
- [ ] Able to set Rating System

### PR DESCRIPTION
**Pull request by AHuangHe**
Made the UI for when a user wants to create a new list.

- [x] Create a user list page/component
- [x] Able to set the Name
- [x] Able to set Rating System

-Values are not saved for now, but once it is connected to the backend, values should be saved,
-Also, the modal might not be needed, but I was thinking that maybe we can add info about the rating system selected on the modal on another issue

Demo

Closes #124

### CODE REVIEW

**Comment by AHuangHe** [comment]
The modal might not be needed, but I was thinking that maybe we can add info about the rating system selected on the modal

**Review by jvoucsb — CHANGES_REQUESTED** [review]
looks good. will need to create an issue to connect to backend in future issue. one small thing and i think thatll be it

**Comment by jvoucsb** [comment]
I think the modal could be a good idea long term to select not only their rating systems but to also search for other rating systems as well and to display for information about a rating system before selecting.

it could also be a good idea to reduce the scope of your issue and keep it simple by only having a dropdown menu of possible rating system ids moving the modal UI in another issue. We will need only the id of the rating system when submitting the form.

File: client/src/components/list/CreateUserList.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    const validationSchema = Yup.object({
+      name: Yup.string()
+        .min(1, 'Name must be at least 1 character long')
+        .max(50, 'Name must be at most 50 characters long')
+        .required('Required'),
+    });
+
+    const formik = useFormik({
+      initialValues:{
+        name: ""
```
[the remark this answers is not in the document]
Comment (AHuangHe) [reply client/src/components/list/CreateUserList.tsx]: I added the rating system field as a string for now

**Comment by AHuangHe** [comment]
Replaced the button to access the modal with a dropdown menu and left the modal for later. I'm not sure how to make the dropdown menu with a variable number of options, so I just left a single temporary option.

File: client/src/components/list/CreateUserList.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        actions.setSubmitting(true);
+
+      }
+    });
+    return(
+        <form onSubmit={formik.handleSubmit}>
+          <Stack spacing={{ base: 8, md: 12 }} mt={4}>
+          <FormControl isRequired>
+            <FormLabel>Name</FormLabel>
+            <Input placeholder="MyUserList" id="name"/>
```
Comment (jvoucsb) [inline client/src/components/list/CreateUserList.tsx:51]: For form inputs make sure to have an `id` prop as well as destructure the `formik.getFieldProps` for the appropriate input

Example: https://github.com/ucsb-cs148-f21/project-t10-animelist/blob/b2c9119256a6d3d3b11459c3bdc3463a0a763a14/client/src/components/forms/LoginForm.tsx#L68

File: client/src/components/list/CreateUserList.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          </FormControl>
+
+          <Stack spacing="24px">
+            <Heading size="md">Choose a Rating System for your List!</Heading>
+            <Select placeholder="---" size="lg" isRequired>
```
Comment (jvoucsb) [inline client/src/components/list/CreateUserList.tsx:56]: Same for this form select here

**Review by jvoucsb — CHANGES_REQUESTED** [review]
Looks good a few changes for the form

File: client/src/components/list/CreateUserList.tsx (hunk 3 above)
Comment (AHuangHe) [reply client/src/components/list/CreateUserList.tsx]: I just added those to both, let me know if it still need changes

**Review by jvoucsb — APPROVED** [review]
good good excellent excellent

**Review by rowantran — APPROVED** [review]
LGTM! great work

**Pull request merged**